cmd/kubeadm/app/cmd/reset.go

```diff
@@ -328,6 +330,15 @@ func resetDetectCRISocket(resetCfg *kubeadmapi.ResetConfiguration, initCfg *kube
 	if resetCfg != nil && len(resetCfg.CRISocket) > 0 {
 		return resetCfg.CRISocket, nil
 	}
+
+	if features.Enabled(initCfg.FeatureGates, features.NodeLocalCRISocket) {
+		kubeletConfig, err := kubeletphase.LoadKubeletConfig(kubeadmconstants.KubeletRunDirectory, kubeadmconstants.KubeletInstanceConfigurationFileName)
+		if err != nil {
+			return "", err
+		}
+		return kubeletConfig.ContainerRuntimeEndpoint, nil
+	}
+
```

Comment:
> this also means all places that construct a InitConfiguration will now have the instance-config.yaml socket applied to noderegistration.crisocket.

I have considered changing this, but it affects too much scope and I am worried that unexpected errors may occur. So in order to reduce the scope of impact, I only changed `reset`. If you think this is reasonable, I will try it in this direction.

Reply:
i think the change has to be made so that we can verify the alpha state of the feature for users.
it's not a complete change if we leave some of the places untouched because of risk.

alpha is disabled by default and the e2e test we need to add will catch problems.
it's also important to get the base feature merged before code freeze and then only send `kind/bug` fixes for it.

you should test kubeadm locally.
- kubeadm init
- kubeadm upgrade apply (to the same version) 

also with kinder if you'd like:
https://github.com/kubernetes/kubeadm/tree/main/kinder

we use it for all our e2e
https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm

Reply:
> you should test kubeadm locally.

Alright, I will implement it in this direction.



> you should test kubeadm locally.

I have already run tests locally before submitting the code, but I am still concerned that there may be some scenarios my tests might not cover.



> we use it for all our e2e
> https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm

I am aware of this; when drafting the KEP, I referred to other KEPs for kubeadm. In the test plan section of the KEP, I mentioned that I will add e2e tests for this.

Reply:
with the FG enabled the general upgrade-1.31->latest test workflow (kinder) should be able to catch any issues. it doesn't not cover all corner cases like phases obviously, but it's still good signal for the alpha.